### ISSUE TITLE
No longer able to run in Minikube-provided Docker environment

### ISSUE THREAD

**Issue by tjensen** [CircleCI-Public/circleci-cli#334]
- [X] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/master/CONTRIBUTING.md).
- [X] I have checked for similar issues and haven't found anything relevant.
- [X] This is not a security issue (which should be reported here: https://circleci.com/security/)

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Running `circleci local execute` results in a `"docker-init": executable file not found` error during the "Spin up Environment" phase:
```
$ minikube version
minikube version: v1.4.0
commit: 7969c25a98a018b94ea87d949350f3271e9d64b6

$ eval $(minikube docker-env)

$ circleci local execute
Docker image digest: sha256:416a81b71713aad8cb82db68637fe3047e9031c4961c1e6a16140f0dff45b8cf
====>> Spin up Environment
Build-agent version 1.0.17052-a0ff09bd (2019-09-30T16:32:40+0000)
Docker Engine Version: 18.09.9
Kernel Version: Linux 38ef40faddbb 4.15.0 #1 SMP Wed Sep 18 07:44:58 PDT 2019 x86_64 Linux
Starting container circleci/python:3.7
  using image circleci/python@sha256:25e42a6d9f3f9f074bdffe119d41935bab8bcca6754c1835cef40a2d35b0969d
Error:
Error response from daemon: exec: "docker-init": executable file not found in $PATH

Step failed
Task failed
Error: Error response from daemon: exec: "docker-init": executable file not found in $PATH
```

I used the following `.circleci/config.yml` file to generate the above error:
```yaml
version: 2
jobs:
  build:
    docker:
      - image: circleci/python:3.7
    working_directory: ~/repo
    steps:
      - checkout
```

**What is the expected behavior?**
The `circleci local execute` command should be able to run in the Docker environment that is provided by [Minikube](https://github.com/kubernetes/minikube).

**Which version of the CLI and OS are you using? Did this work in previous versions?**

```
$ circleci version
0.1.5879+416032d
```

```
$ circleci diagnostic

---
CircleCI CLI Diagnostics
---
Debugger mode: false
Config found: /Users/tjensen/.circleci/cli.yml
API host: https://circleci.com
API endpoint: graphql-unstable
Error: please set a token with 'circleci setup'
You can create a new personal API token here:
https://circleci.com/account/api
```

I'm running the CLI on macOS Mojave version 10.14.6.

I used to be able to run the `circleci` CLI with Minikube when I was using version `0.1.4786+bad101f`, but I have since upgraded to the latest version (using `brew upgrade circleci`). I have tried manually reinstalling the older version but it's still failing with the `docker-init` error. I presume it has something to do with the Build-agent but I'm not sure how to force an older version of the agent. I found an old saved log from a working run that showed I was using version `0.1.750-336e0196` of the build agent in Minikube.

**Comment by tjensen** [comment]
I was able to force the CLI to use an older version of the build agent by doing the following:
1. Deleting the `circleci/picard` image from my Docker environment:
```
$ docker rmi circleci/picard:latest
```
2. Editing my `~/.circleci/build_agent_settings.json` file to contain the older build agent's image sha256 hash:
```json
{"LatestSha256":"sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08"}
```

Now, when I re-run the example, it completes successfully:
```
$ circleci local execute
Docker image digest: sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08
Unable to find image 'circleci/picard@sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08' locally
sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08: Pulling from circleci/picard
4fe2ade4980c: Already exists
1cd63deee868: Pull complete
345f9fa541d7: Pull complete
273ae9fccbcf: Pull complete
e2fd7699b0fd: Pull complete
4be3a1acec01: Pull complete
Digest: sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08
Status: Downloaded newer image for circleci/picard@sha256:3d88cfffd448da07ff8ced22ccdca5adc09ff18e87dc7db9b8af5b8a47647c08
====>> Spin up Environment
Build-agent version 0.1.750-336e0196 (2018-09-13T19:48:06+0000)
Starting container circleci/python:3.7
  using image circleci/python@sha256:25e42a6d9f3f9f074bdffe119d41935bab8bcca6754c1835cef40a2d35b0969d

Using build environment variables:
  BASH_ENV=/tmp/.bash_env-localbuild-1570025547
  CI=true
  CIRCLECI=true
  CIRCLE_BRANCH=
  CIRCLE_BUILD_NUM=
  CIRCLE_JOB=build
  CIRCLE_NODE_INDEX=0
  CIRCLE_NODE_TOTAL=1
  CIRCLE_REPOSITORY_URL=
  CIRCLE_SHA1=
  CIRCLE_SHELL_ENV=/tmp/.bash_env-localbuild-1570025547
  CIRCLE_WORKING_DIRECTORY=~/repo

====>> Checkout code
  #!/bin/bash -eo pipefail
mkdir -p /home/circleci/repo && cp -r /tmp/_circleci_local_build_repo/. /home/circleci/repo
Success!
```

While this gets me unblocked, I don't consider it a fix because I'm unable to upgrade the build agent for bug fixes, etc.

**Comment by tjensen** [comment]
It looks like this issue was fixed in Minikube v1.6.0 by https://github.com/kubernetes/minikube/pull/5818
